src/backends/backend/python/resources/utils.py

```diff
@@ -0,0 +1,160 @@
+    and simultaneously calling different methods with different threads is
+    not supported and will cause undefined behavior.
+
+    Parameters
+    ----------
+    inputs : list
+        A list of Tensor objects, each describing data for an input tensor
+        required by the model
+    request_id : str
+        Optional identifier for the request. If specified will be returned
+        in the response. Default value is an empty string which means no
+        request_id will be used.
+    correlation_id : str
+        Optional identifier for the request. If specified will be returned
+        in the response. Default value is an empty string which means no
+        request_id will be used.
+    requested_output_name : list
+        A list of strings, each describing the requested output.
+    """
+
+    def __init__(self, inputs, request_id, correlation_id,
+                 requested_output_names):
+        self._inputs = inputs
+        self._request_id = request_id
+        self._correlation_id = correlation_id
+        self._requested_output_names = requested_output_names
+
+    def inputs(self):
+        """Get input tensors
+        Returns
+        ----
+        list
+            A list of input Tensor objects
+        """
+        return self._inputs
+
+    def request_id(self):
+        """Get request ID
+        Returns
+        -------
+        str
+            Request ID
+        """
+        return self._request_id
+
+    def correlation_id(self):
+        """Get correlation ID
+        Returns
+        -------
+        int
+            Request correlation ID
+        """
+        return self._correlation_id
+
+    def requested_output_names(self):
+        """Get requested output names
+        Returns
+        -------
+        list
+            A list of strings, each describing the requested output name
+        """
+        return self._requested_output_names
+
+
+class InferenceResponse:
+    """An InfrenceResponse object is used to represent the response to an
+    inference request. The object is intended to be used by a single thread
+    and simultaneously calling different methods with different threads is
+    not supported and will cause undefined behavior.
+
+    Parameters
+    ----------
+    output_tensors : list
+        A list of Tensor objects, each describing data for an output tensor
+        required the InferenceRequest
+    """
+
+    def __init__(self, output_tensors):
+        self._output_tensors = output_tensors
+
+    def output_tensors(self):
+        """Get output tensors
+        Returns
+        -------
+        list
+            A list of Tensor objects
+        """
+        return self._output_tensors
+
+
+class Tensor:
+    """A Tensor object is used to represent inputs and output data for an
+    InferenceRequest or InferenceResponse. The object is intended to be used
+    by a single thread and simultaneously calling different methods with
```

Comment:
These thread comments are not really correct. These are trivial classes so I don't think  you need to say anything about threading.